src/ddscxx/include/dds/topic/detail/ContentFilteredTopic.hpp

```diff
@@ -275,6 +275,9 @@ class ContentFilteredTopic  :
                 /* Empty: the exception throw should have already traced an error. */
             }
         }
+        // delete the functor
+        delete myFunctor;
+        myFunctor = nullptr;
```

Comment:
This fixes a leak with the filter function. As we didn't have tests with the content-filtered topics, we didn't catch this!

Reply:
Good catch!